### PR TITLE
feat: add basic spell checker

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: crafters/nlp/PySpellChecker/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+manifest_version: 1
+name: PySpellChecker
+kind: crafter
+description: Does word by word correction of input texts
+author: Jina AI Dev-Team (dev-team@jina.ai)
+url: https://jina.ai
+vendor: Jina AI Limited
+documentation: https://github.com/jina-ai/jina-hub/tree/master/crafters/nlp/PySpellChecker/README.md
```
Comment (JoanFM) [inline crafters/nlp/PySpellChecker/manifest.yml:8]:
```suggestion
documentation: https://github.com/jina-ai/jina-hub/blob/master/crafters/nlp/PySpellChecker/README.md
```

File: crafters/nlp/PySpellChecker/__init__.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            not, only available when not using a language dictionary.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+
+    """
+
+    def __init__(self,
+                 language: str = 'en',
+                 local_dictionary: Optional[str] = None,
+                 distance: int = 2,
```
Comment (bwanglzu) [inline crafters/nlp/PySpellChecker/__init__.py:36]: maybe it's better to rename `distance` to `edit_distance`?

File: crafters/nlp/PySpellChecker/__init__.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        Craft sentences correcting misspelled words
+
+        :param text: The text to be corrected
+        :param args:  Additional positional arguments
+        :param kwargs: Additional keyword arguments
+        :return: A dictionary with the extracted text
+        """
+        words = self.speller.split_words(text)
+        corrected_text = ' '.join([self.speller.correction(word) for word in words])
+        return dict(text=corrected_text)
```
Comment (bwanglzu) [inline crafters/nlp/PySpellChecker/__init__.py:65]: maybe we can add a static method to list supported languages

```suggestion
@staticmethod
def list_supported_languages():
    return ['en', 'es' ...]
```

**Review by Kelton8Z — APPROVED** [review]
LGTM

**Pull request merged**